reconcile/openshift_prometheusrules.py

```diff
@@ -0,0 +1,273 @@
+import json
+import semver
+import logging
+import traceback
+
+import utils.gql as gql
+import reconcile.openshift_base as ob
+import utils.jsonnet as jsonnet
+import utils.template as template
+
+from utils.defer import defer
+from utils.openshift_resource import (OpenshiftResource as OR)
+
+SLO_RULES = 'slo-rules.jsonnet.j2'
+
+PERFORMANCE_PARAMETERS_QUERY = """
+{
+  performance_parameters_v1 {
+    labels
+    name
+    component
+    prometheusLabels
+    namespace {
+      name
+      cluster {
+        observabilityNamespace {
+          name
+          cluster {
+            name
+            serverUrl
+            jumpHost {
+              hostname
+              knownHosts
+              user
+              port
+              identity {
+                path
+                field
+                format
+              }
+            }
+            automationToken {
+              path
+              field
+              format
+            }
+            disable {
+              integrations
+            }
+          }
+        }
+        name
+        serverUrl
+        jumpHost {
+          hostname
+          knownHosts
+          user
+          port
+          identity {
+            path
+            field
+            format
+          }
+        }
+        automationToken {
+          path
+          field
+          format
+        }
+        internal
+        disable {
+          integrations
+        }
+      }
+    }
+    sloRules {
+      name
+      kind
+      metric
+      percentile
+      selectors
+      httpStatusLabel
+    }
+    volume {
+      name
+      threshold
+      rule
+      additionalLabels
+    }
+    availability {
+      name
+      additionalLabels
+      rules {
+        latency
+        errors
+      }
+    }
+    latency {
+      name
+      threshold
+      rule
+      additionalLabels
+    }
+    errors {
+      name
+      target
+      rule
+      additionalLabels
+    }
+    rawRecording {
+      record
+      expr
+      labels
+    }
+  }
+}
+"""
+
+QONTRACT_INTEGRATION = 'openshift-prometheusrules'
+QONTRACT_INTEGRATION_VERSION = semver.format_version(0, 1, 0)
+
+
+def labels_to_selectors(labels):
+    return ", ".join(['\'%s="%s"\'' % (k, v) for k, v in labels.items()])
```

Comment:
can we use f-strings instead of the py2 formatting?
here and everywhere else as well...

Reply:
will do

Reply:
Fixed in 83bbea3